docs/spec/nmt-wrapper.md

```diff
@@ -0,0 +1,83 @@
+# Namespaced Merkle Tree Wrapper
+## Abstract
+In Celestia, block transactions are grouped into identical-size shares and arranged in a `k` by `k` matrix, called [original data square](https://github.com/celestiaorg/celestia-app/blob/specs-staging/specs/src/specs/data_structures.md#arranging-available-data-into-shares).
```

Comment:
> identical-size shares

Do we want to call out which layer owns defining that size? 

A related question that may be for just for my understanding, what, if any, are the notable impacts to changing the share size?

Reply:
> what, if any, are the notable impacts to changing the share size?

Relevant context: https://github.com/celestiaorg/celestia-app/issues/711

Shares are the unit that gets downloaded by light clients during DAS so:
- Increasing the share size increases the amount of data a light client must download per sample
- Decreasing the share size decreases the amount of data a light client must download per sample

Each share has a fixed size prefix (e.g. namespace ID, info byte) and some variable prefix depending on the type of the share so:
- Increasing the share size decreases the % of overhead occupied by the prefix. 
- Decreasing the share size increases the % of overhead occupied by the prefix.

Additionally if all other variables remain constant (i.e. max square size):
- Increasing the share size increases the capacity for data in the data square
- Decreasing the share size decreases the capacity for data in the data square

Reply:
> Do we want to call out which layer owns defining that size?

AFAIK, the maximum value of `k` is governed by the [`AVAILABLE_DATA_ORIGINAL_SQUARE_MAX`](https://github.com/celestiaorg/celestia-app/blob/c09843d07d4c3842753138de96b304b4866e8f5d/specs/src/specs/consensus.md#constants) constant among other Celestia consensus rule constants. However, the actual value of `k` depends on the number of shares in each block and is the closest power of 2 to a square enclosing all the shares.
Quoted from [Header description](https://github.com/celestiaorg/celestia-app/blob/c09843d07d4c3842753138de96b304b4866e8f5d/specs/src/specs/data_structures.md#header)
> The size of the [original data square](https://github.com/celestiaorg/celestia-app/blob/c09843d07d4c3842753138de96b304b4866e8f5d/specs/src/specs/data_structures.md#arranging-available-data-into-shares), availableDataOriginalSquareSize, isn't explicitly declared in the block header. Instead, it is implicitly computed as the smallest power of 2 whose square is at least availableDataOriginalSharesUsed (in other words, the smallest power of 4 that is at least availableDataOriginalSharesUsed).

Reply:
> Do we want to call out which layer owns defining that size?

Apologies for the confusion. In my previous response, I misunderstood your question and provided information about the `k` value. To clarify, the share size in Celestia is also determined by a constant called `SHARE_SIZE`, which is defined as part of the Celestia consensus rule (AFAIK).

Reply:
Added clarification about the share size and `k` value in this update https://github.com/celestiaorg/celestia-app/pull/1442/commits/0ad159a7d60d0fdcf7b560d3796b4ed6f68cf9e8.

Reply:
sounds like this similarly applies more elsewhere, so I'll resolve.